pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -121,15 +123,18 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetad
         this.sortedColumn = null;
       }
     }
-    //inverted index columns
-    invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-    if (sortedColumn != null && !invertedIndexColumns.contains(sortedColumn)) {
+
+    // Inverted index columns
+    Set<String> invertedIndexColumns = indexLoadingConfig.getInvertedIndexColumns();
+    if (sortedColumn != null) {
```

Comment:
Find the reason. The motivation is that, when we seal the realtime in memory segment into offline segment, we use sortedColumn's inverted index to maintain the order of records. Will add comments.